lib/tasks/pipeline.js

```diff
@@ -0,0 +1,100 @@
+var Task        = require('ember-cli/lib/models/task');
+var Promise     = require('ember-cli/lib/ext/promise');
+var SilentError = require('ember-cli/lib/errors/silent');
+
+var existsSync  = require('fs').existsSync;
+var path        = require('path');
+var dotenv      = require('dotenv');
+
+var Pipeline = require('../utilities/pipeline');
+
+module.exports = Task.extend({
+  init: function() {
+    if (!this.project) {
+      throw new SilentError('No project passed to pipeline task');
+    }
+
+    if (!this.ui) {
+      throw new SilentError('No ui passed to pipeline task');
+    }
+
+    this._pipeline = this.pipeline || new Pipeline(this.hooks, {
+      ui: this.ui
+    });
+
+    this.project.addons.forEach(this._registerPipelineHooks.bind(this));
+
+    var root = this.project.root;
+
+    this.deployEnvironment = this.deployEnvironment || 'production';
+
+    var dotEnvFilename = '.env.deploy.' + this.deployEnvironment;
+    var dotEnvFilePath = path.join(root, dotEnvFilename);
+
+    if (existsSync(dotEnvFilePath)) {
+      dotenv.load({
```

Comment:
Loading env vars from .env.deploy.<environment> files if they exist.  I attempted to create an ember-cli-deploy-dotenv plugin but this wasn't really possible.  Happy to discuss why if needed.  Seems that this is a pretty standard requirement so I think it makes sense to do it in the PipelineTask.